packages/svelte-docgen/src/doc/type.ts

```diff
@@ -49,6 +50,12 @@ export type Type =
 	| TypeParam
 	| Union;
 
+export type TypeOrRef = Type | TypeRef;
+
+export function isTypeRef(type?: TypeOrRef): type is TypeRef {
```

Comment:
```suggestion
/**
 * A type-guard to check if the type is a _reference_ to an aliased or named type stored in {@link Types}. The reason we have it isolated in separate storage _(`types`)_ is to avoid circularity.
 * 
 * @example Basic usage in real-word scenario
 * 
 * ```js
 * import { isTypeRef } from "svelte-docgen/doc";
 * 
 * const { props, types } = parse("... <Svelte code> ...");
 * //             👆 stored aliased/named types
 * const exampleProp = prop.get("example");
 * 
 * if (isTypeRef(exampleProp.type)) {
 *   const examplePropType = types[exampleProp.type]; // 👈 This is how you can access the type when it's a `TypeRef`.
 * }
 * ```
 */
export function isTypeRef(type?: TypeOrRef): type is TypeRef {
```

Reply:
Crap, I just realized that this module is `.ts` file. The reason I made it was to provide a module with _exportable_ types. 😓